homeassistant/components/api/__init__.py

```diff
@@ -410,7 +410,7 @@ async def post(self, request):
         try:
             data = await request.json()
             tpl = template.Template(data["template"], request.app["hass"])
-            return tpl.async_render(data.get("variables"))
+            return str(tpl.async_render(data.get("variables")))
```

Comment:
Why convert to string?

Reply:
Maybe make string conversion default now with option to disable it via query. Later drop that.

Reply:
Because the response on the API is expected to be a string.

Reply:
> Maybe make string conversion default now with option to disable it via query. Later drop that.

It will always be transferred as string, as it is simply HTTP body contents.

Reply:
Ooh, we don't return JSON here, right 🤦 that's a failure of API design

Reply:
Well before this PR (the current situation), it will always have been a string anyways. With this PR, types all of sudden become a possibility (including a structured response like JSON).